### PR TITLE
AO3-3546 change abuse form

### PR DESCRIPTION
**Pull request by Talvalin**
* Modified AbuseReport model to include a metadata attribute which contains the title and the author of the work, when a work is reported
* Modified the AdminMailer and UserMailer to pull the author and title from the metadata and pass it to the AbuseReport mail views
* Modified abuse mailer spec to check for correct metadata
* Added migration required to add the metadata field to the AbuseReport ActiveRecord

### CODE REVIEW

**Comment by Talvalin** [comment]
Jira link: https://otwarchive.atlassian.net/browse/AO3-3546

File: app/views/admin_mailer/abuse_report.html.erb [hunk 1]
```diff
@@ -4,6 +4,9 @@
   
   Email of the person who reported the abuse, if given: <%= style_email(@email, @email) %><br>
   URL of the page the abuse is on: <%= style_link(@url, @url) %><br>
+  
+  Title of the work: <%= @title.html_safe %><br/>
+  Author(s) of the work: <%= @author.html_safe %><br/>
```
Comment (sarken) [inline app/views/admin_mailer/abuse_report.html.erb]: Drive-by front end nitpick: to be less text-centric, we're using "creator" instead of "author" in any interface/email/etc text now.

**Comment by ariana-paris** [comment]
Unfortunately, the test failures Travis is reporting are real failures. The Abuse tests are failing in both the Rspec and Cucumber tests, so these will need to be fixed before this can go in.

File: app/views/admin_mailer/abuse_report.html.erb [hunk 2]
```diff
@@ -4,6 +4,11 @@
   
   Email of the person who reported the abuse, if given: <%= style_email(@email, @email) %><br>
   URL of the page the abuse is on: <%= style_link(@url, @url) %><br>
+  
+  <% if @has_metadata %>
+    Title of the work: <%= @title.html_safe %><br/> 
+    Author(s) of the work: <%= @author.html_safe %><br/>
```
Comment (sarken) [inline app/views/admin_mailer/abuse_report.html.erb]: Whoops, looks like my last comment got eaten by updates. The emails should say "Creator(s)" rather than "Author(s)" because we're trying to move away from text-centric language.

**Comment by Talvalin** [comment]
My understanding was that it's not possible to run tests on Webdevs, so is
there any way that I can verify a test before pushing my commits?

File: spec/mailers/abuse_mailer_spec.rb [hunk 3]
```diff
@@ -87,11 +87,44 @@
...
+   
    it "has the url of the page with abuse" do
      expect(mail.html_part).to have_body_text(report.url)
    end
   end
 
+  
+  context "abuse_reports contains creator and title if reporting a work" do
+   let(:work) {create(:work)}
+   let(:report) {create(:abuse_report, url: work_url(work.url)}
```
Comment (ariana-paris) [inline spec/mailers/abuse_mailer_spec.rb]: What Hound is trying to say is that you have a missing bracket here - your `create()` call is missing its closing bracket. That line should be:
```
  let(:report) {create(:abuse_report, url: work_url(work.url))}
```

File: spec/mailers/abuse_mailer_spec.rb [hunk 4]
```diff
@@ -87,11 +87,44 @@
...
+   end
+
+   it "body text contains the comment" do
+     expect(mail).to have_body_text(/#{report.comment}/)
+   end
+   
+   it "has the url of the page with abuse" do
+     expect(mail.html_part).to have_body_text(report.url)
+   end
+  end
```
Comment (ariana-paris) [inline spec/mailers/abuse_mailer_spec.rb]: It's hard to tell in the diff, but this means you probably have too many `end`s or your `do...end` blocks aren't properly matched.

**Comment by sarken** [comment]
Hihi, just leaving a note for anyone who reviews or merges this: the migration should be good to go because it is 20150907163906, and our schema is at 20150901132832

**Comment by zz9pzza** [comment]
Sorry this has gone grey :(

**Comment by sarken** [comment]
Hi!

Just a note that I'm closing this pull request and issue because Abuse will be switching to a new ticket tracker soon, so this code will no longer work. Additionally, Abuse isn't sure if they're going to want this feature for the new form due to an uptick in reports for things that aren't works.

Thank you for working on this, though!
